File: oo-workshop/single-responsibillity-principle/src/test/java/com/example/interfacesegregation/mobprogramming/problem/WalkInClientImplTest.java
package com.example.interfacesegregation.mobprogramming.problem;

import org.junit.Test;;

import static org.junit.Assert.assertTrue;

public class WalkInClientImplTest {
    @Test
    void walkInCustomerOrder(){
        //arrange
        WalkInClientImpl given = new WalkInClientImpl("Roger Miller", "Cheese Sandwich", "9.99");

        //act
        String when = given.walkInCustomerOrder();

        //assert
        assertTrue(when.contains("Roger Miller"));
        assertTrue(when.contains("Cheese Sandwich"));
    }

    @Test
    void payInPerson(){
        //arrange
        WalkInClientImpl given = new WalkInClientImpl("Roger Miller", "Cheese Sandwich", "9.99");

        //act
        String when = given.payInPerson();

        //assert
        assertTrue(when.contains("9.99"));
    }
}

File: oo-workshop/tsexercises/tests/unit/interfacesegregation/walkInClientImpl.test.ts
import {WalkInClientImpl} from '../../../src/interfacesegregation/walkInClientImpl'

describe('walk in client implementation', () => {
 
    it('should accept a walk in order', () => {
        //arrange
        const given: WalkInClientImpl = new WalkInClientImpl("Roger Miller", "sand", "9.99")

        //act
        const when: string = given.walkInCustomerOrder()

        //assert
        expect(when).toContain("Roger Miller")
        expect(when).toContain("sand")
    })


    it('should accept in person payment', () => {
        //arrange
        const given: WalkInClientImpl = new WalkInClientImpl("Roger Miller", "sand", "9.99")

        //act
        const when: string = given.payInPerson()

        //assert
        expect(when).toContain("9.99")
    })
})

